src/test_bst.py

```diff
@@ -10,6 +10,24 @@ def empty_tree():
     return BinarySearchTree()
 
 
+@pytest.fixture
+def tree_init_list():
+    """Init with numbers 1-7 tree fixture."""
+    return BinarySearchTree([4, 2, 3, 1, 6, 5, 7])
```

Comment:
Your fixtures are good but you should add another that is an unbalanced tree, to make sure in the strangest trees the traversals still work.